src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -751,54 +755,105 @@ private void transferAlignmentInfoToPairedRead(final SAMRecord firstUnaligned, f
     }
 
     /**
-     * Checks to see whether the ends of the reads overlap and soft clips reads
-     * them if necessary.
+     * Checks to see whether the ends of the reads overlap and clips reads
+     * if necessary.
      */
-    protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRecord read2) {
+    protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRecord read2, final CigarOperator clippingOperator) {
```

Comment:
I think it's better to use a boolean as there are only 2 resaonable operators here....SOFT and HARD.

Reply:
(and then the sanity checks below will not be needed)